src/gui/guiAnimatedImage.cpp

```diff
@@ -0,0 +1,33 @@
+#include "guiAnimatedImage.h"
+
+#include "client/guiscalingfilter.h"
+#include "client/tile.h" // ITextureSource
+#include "client/texture_pool.h"
+#include "log.h"
+
+GUIAnimatedImage::GUIAnimatedImage(gui::IGUIEnvironment *env, gui::IGUIElement *parent,
+		s32 id, const core::rect<s32> &rectangle, const std::string &name,
+		ISimpleTextureSource *tsrc, TexturePool *pool) :
+		gui::IGUIElement(gui::EGUIET_ELEMENT, env, parent, id, rectangle),
+		m_name(name), m_texture_idx(0), m_texture_pool(pool), m_tsrc(tsrc)
+{
+}
+
+void GUIAnimatedImage::draw()
+{
+	video::ITexture *texture = m_texture_pool->getTexture(m_name, m_tsrc, m_texture_idx);
+
+	if (texture) {
+		video::IVideoDriver *driver = Environment->getVideoDriver();
+
+		const video::SColor color(255, 255, 255, 255);
+		const video::SColor colors[] = {color, color, color, color};
+
+		draw2DImageFilterScaled( driver, texture, AbsoluteRect,
+				core::rect<s32>(core::position2d<s32>(0, 0), texture->getOriginalSize()),
+				NULL, colors, true);
+	} else {
+		errorstream << "GUIAnimatedImage::draw() unable to load texture:" << std::endl;
+		errorstream << "\t" << m_name << std::endl;
```

Comment:
Error message spam in draw functions can be problematic.

Reply:
I just ported that log over from the original code. Would you prefer I remove it altogether, or should it be moved somewhere specific?

Reply:
I don't know. Finding this error elsewhere (eg, in the parser) will probably be difficult.

Reply:
Actually... when the file isn't there it will normally generate a replacement and log an error anyway. I've decided to remove it, but I'll leave this unresolved in case others disagree.

Reply:
Since an error is still logged with the logs removed altogether and no-one is piping up about this, I'm marking this as resolved.